clion/src/191/main/kotlin/org/rust/clion/valgrind/RsValgrindConfigurationExtension.kt

```diff
@@ -0,0 +1,194 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.clion.valgrind
+
+import com.intellij.execution.ExecutionException
+import com.intellij.execution.configurations.CommandLineState
+import com.intellij.execution.configurations.GeneralCommandLine
+import com.intellij.execution.configurations.RunnerSettings
+import com.intellij.execution.configurations.SearchScopeProvider
+import com.intellij.execution.process.ProcessAdapter
+import com.intellij.execution.process.ProcessEvent
+import com.intellij.execution.process.ProcessHandler
+import com.intellij.execution.runners.ExecutionEnvironment
+import com.intellij.execution.ui.ConsoleView
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.options.ShowSettingsUtil
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.Disposer
+import com.intellij.openapi.util.Key
+import com.intellij.openapi.util.SystemInfo
+import com.intellij.openapi.util.io.FileUtil
+import com.intellij.openapi.util.text.StringUtil
+import com.intellij.util.ArrayUtil
+import com.intellij.util.ui.StatusText
+import com.jetbrains.cidr.cpp.CPPBundle
+import com.jetbrains.cidr.cpp.profiling.*
+import com.jetbrains.cidr.cpp.profiling.ui.MemoryProfileOutputPanel
+import com.jetbrains.cidr.cpp.valgrind.*
+import com.jetbrains.cidr.lang.toolchains.CidrToolEnvironment
+import org.rust.cargo.runconfig.CargoCommandConfigurationExtension
+import org.rust.cargo.runconfig.ConfigurationExtensionContext
+import org.rust.cargo.runconfig.command.CargoCommandConfiguration
+import org.rust.cargo.runconfig.console.CargoConsoleBuilder
+import java.io.File
+import java.io.IOException
+
+private val LOG = Logger.getInstance(RsValgrindConfigurationExtension::class.java.name)
+
+class RsValgrindConfigurationExtension : CargoCommandConfigurationExtension() {
+    override fun isApplicableFor(configuration: CargoCommandConfiguration): Boolean {
+        return true
+    }
+
+    override fun isEnabledFor(applicableConfiguration: CargoCommandConfiguration, runnerSettings: RunnerSettings?): Boolean =
+        SystemInfo.isLinux || SystemInfo.isMac
```

Comment:
Maybe we want to enable it only on Linux?
CC @Undin

Reply:
It's enabled for macOS for C++ so I think it's ok for us too even it works not perfect